src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -129,16 +157,32 @@ public async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobReco
             using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
             using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
             {
+                async void OnRetry(Exception exception, TimeSpan timeSpan, int i, Context context)
+                {
+                    _pollyRetryLoggerFactory.Create();
```

Comment:
We still want to log, right? So we'll need to use the resulting delegate. Furthermore, can we resolve the delegate earlier in the ctor as opposed to every call?

Reply:
As a follow up to our discussion, I confirmed that `Create()` returns a logging Action:


```
public Action<Exception, TimeSpan, int, Context> Create()
{
    ILogger logger = _loggerFactory.CreateLogger("PollyRetryLogger");

    return (exception, sleepDuration, retryCount, context) =>
    {
        LogRetryDelegate(logger, sleepDuration, retryCount, exception);
    };
}
```

The `LogRetryDelegate` is defined as:

```
private static readonly Action<ILogger, TimeSpan, int, Exception> LogRetryDelegate =
  LoggerMessage.Define<TimeSpan, int>(
      LogLevel.Warning,
      default,
      "The operation failed. Will retry in '{SleepDuration}'. Retried {RetryCount} of time(s) so far.");
```

In `SqlServerTransientFaultRetryPolicyFactory` (the class that creates the default retry policy), we set the `OnRetry` action to `LogRetryDelegate`. If we were to move forward with this PR, we could avoid passing in `PollyRetryLoggerFactory` all together by adding a logging call to the custom `OnRetry` Actions.